### PR TITLE
Outfit variant support and Freelancer variants

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
This PR changes job selection code to randomly choose any outfit associated with that particular job. Currently, each job only has one set outfit.

More as a proof of concept if anything, I have applied this to the Free Lancer ERT, building on Hyper's recent work there. I have also added a new "veteran" job for freelancers, as a kind of fire support role. Basically they get a (debateably) better gun.

Three leader variants
1. Existing M16 dude
2. TX11 tac sensor man
3. TX55 chad

Three standards variants
1. Existing M16 dude
2. M16 dude with UGL instead of under shotty
3. TX11 man

Three veteran variants
1. ALF run and gun dude
2. M412L1 machine gunner
3. TX55 chad

Tivi said TX55 is OK for ERT. It is the original variation of the TX54, with an underbarrel rifle component. Should be rad.

## Why It's Good For The Game
Variation is cool, and it lays the groundwork to use to else where for events/game modes/whatever.

## Changelog
:cl:
add: Freelancers in the sector have been reported wielding a wider variety of weapons
code: Support added for jobs to have multiple outfits to randomly select from
/:cl:

### CODE REVIEW

File: code/datums/jobs/job/job.dm [hunk 1]
```diff
@@ -284,8 +285,10 @@ GLOBAL_PROTECT(exp_specialmap)
 				stack_trace("[src] had an ID when apply_outfit_to_spawn() ran")
 			QDEL_NULL(wear_id)
 		equip_to_slot_or_del(id_card, SLOT_WEAR_ID)
+		var/datum/outfit/variant = pick(typesof(job.outfit)) ///probs can just add the new statement here and remove the line below?
+		variant = new variant
```
Comment (TiviPlus) [inline code/datums/jobs/job/job.dm:289]: whats up with the comment hefre

File: code/modules/clothing/modular_armor/attachments/storage.dm [hunk 2]
```diff
@@ -111,13 +111,28 @@
...
 
-/obj/item/armor_module/storage/ammo_mag/freelancer/Initialize()
+/obj/item/armor_module/storage/ammo_mag/m16/Initialize()
 	. = ..()
 	new /obj/item/ammo_magazine/rifle/m16(storage)
 	new /obj/item/ammo_magazine/rifle/m16(storage)
 	new /obj/item/ammo_magazine/rifle/m16(storage)
 	new /obj/item/ammo_magazine/rifle/m16(storage)
 
+/obj/item/armor_module/storage/ammo_mag/tx11/Initialize()
```
Comment (TiviPlus) [inline code/modules/clothing/modular_armor/attachments/storage.dm:121]: no nunbers in typepaths please

File: code/modules/clothing/modular_armor/attachments/uniform.dm [hunk 3]
```diff
@@ -157,6 +157,10 @@
 	. = ..()
 	new /obj/item/weapon/gun/pistol/g22(storage)
 
+/obj/item/armor_module/storage/uniform/holster/vp70/Initialize()
+	. = ..()
+	new /obj/item/weapon/gun/pistol/vp70(storage)
```
Comment (TiviPlus) [inline code/modules/clothing/modular_armor/attachments/uniform.dm:162]: ditto

File: code/modules/clothing/suits/marine_armor.dm [hunk 4]
```diff
@@ -652,11 +652,31 @@
...
 	)
 	starting_attachments = list(
 		/obj/item/armor_module/module/valkyrie_autodoc,
-		/obj/item/armor_module/storage/ammo_mag/freelancer,
+		/obj/item/armor_module/storage/ammo_mag/m16,
+	)
+
+/obj/item/clothing/suit/storage/faction/freelancer/leader/tx11
+	attachments_allowed = list(
+		/obj/item/armor_module/module/valkyrie_autodoc,
```
Comment (TiviPlus) [inline code/modules/clothing/suits/marine_armor.dm:664]: ditto

File: code/modules/clothing/under/marine_uniform.dm [hunk 5]
```diff
@@ -257,6 +257,9 @@
 	has_sensor = 0
 	starting_attachments = list(/obj/item/armor_module/storage/uniform/holster/freelancer)
 
+/obj/item/clothing/under/marine/veteran/freelancer/Veteran
+	starting_attachments = list(/obj/item/armor_module/storage/uniform/holster/vp70)
```
Comment (TiviPlus) [inline code/modules/clothing/under/marine_uniform.dm:261]: snake_case not CamelCase pls
```suggestion
/obj/item/clothing/under/marine/veteran/freelancer/veteran
	starting_attachments = list(/obj/item/armor_module/storage/uniform/holster/vp70)
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest is ok

File: code/datums/jobs/job/job.dm (hunk 1 above)
Comment (Lumipharon) [reply code/datums/jobs/job/job.dm]: whoops, I chuck in notes to myself as I write. Removed.

File: code/modules/clothing/modular_armor/attachments/storage.dm (hunk 2 above)
Comment (Lumipharon) [reply code/modules/clothing/modular_armor/attachments/storage.dm]: done

File: code/modules/clothing/suits/marine_armor.dm (hunk 4 above)
Comment (Lumipharon) [reply code/modules/clothing/suits/marine_armor.dm]: dome

File: code/modules/clothing/modular_armor/attachments/uniform.dm (hunk 3 above)
Comment (Lumipharon) [reply code/modules/clothing/modular_armor/attachments/uniform.dm]: dome

**Comment by Lumipharon** [comment]
@TiviPlus changes made.

**Review by TiviPlus — APPROVED** [review]

**Comment by lKiseki** [comment]
Headsets are breaking hard, for all jobs. Roundstart synths, ROs, latejoins even; all headsets seem to not be working as intended, with varying frequencies and no encryption keys on any of them.

**Pull request merged**